### PR TITLE
Refactor `count_occurrences`

### LINKED ISSUE #1073: Refactor `count_occurrences` to no longer use `make_afun`
Refactor `count_occurrences` and `a_count_occurrences` to no longer require `make_afun` and allow for greater customizability.

### PR DESCRIPTION
**Pull request by edelarua**
# Pull Request

Fixes #1073

### CODE REVIEW

**Comment by Melkiades** [comment]
@shajoezhu I think we can put this one on hold for this OCEAN release, what do you think @edelarua?

**Comment by edelarua** [comment]
> @shajoezhu I think we can put this one on hold for this OCEAN release, what do you think @edelarua?

I agree - this can be put in the backlog until we have need for it/have some downtime for review.

File: NEWS.md [hunk 1]
```diff
@@ -8,6 +8,10 @@
 ### New Features
 * Added the `na_str` argument to `analyze` & `summarize_row_groups` wrapper functions `count_abnormal`, `count_abnormal_by_baseline`, `count_abnormal_by_marked`, `count_abnormal_by_worst_grade`, `count_abnormal_lab_worsen_by_baseline`, `count_cumulative`, `count_missed_doses`, `count_occurrences`, `count_occurrences_by_grade`, `summarize_occurrences_by_grade`, `summarize_patients_events_in_cols`, `count_patients_with_event`, `count_patients_with_flags`, `count_values`, `estimate_multinomial_response`, `estimate_proportion`, `h_tab_one_biomarker`, `estimate_incidence_rate`, `logistic_summary_by_flag`, `estimate_odds_ratio`, `estimate_proportion_diff`, `test_proportion_diff`, `summarize_ancova`, `summarize_change`, `summarize_glm_count`, `summarize_num_patients`, `analyze_num_patients`, `summarize_patients_exposure_in_cols`, `coxph_pairwise`, `tabulate_survival_subgroups`, `surv_time`, and `surv_timepoint`.
 
+### New Features
+* Added function `get_indents_from_stats` to format and return indent modifiers for a given set of statistics.
+* Added summarize function version of `count_occurrences` analyze function, `summarize_occurrences`.
```
Comment (Melkiades) [inline NEWS.md:13]: to move up

**Review by Melkiades — APPROVED** [review]
Looks amazing Emily! Thank you. I am a bit confused by row_nms though: is it for columns actually? or different row groups? This behavior is a bit difficult to surface to the user, isn't it? Just wondering ^^

**Comment by edelarua** [comment]
Merge after https://github.com/insightsengineering/chevron/pull/688 to avoid any downstream changes in chevron.

**Pull request merged**